### ISSUE TITLE
Codec doesn't show up anywhere, install failed without errors?

### ISSUE THREAD

**Issue by GunStingerNL** [jackun/TestAMFVFW#13]
I've been trying to install the codec for the past 3 hours now, but I have so far been unable to get it to show up anywhere. Running Windows 7 with an R9 280X, driver version 16.8.2

I unpacked the whole thing into C:\TestAMFVFW-master, manually unblocked the main files (install.bat, the .inf and .sln and both .dlls), ran install.bat as admin, aaaand nothing. No errors, but the codec just doesn't show up anywhere. Not in DXTory, not in Vegas, not in Handbrake, and not in Sherlock. I have also reinstalled every version of VC++ from 2013 onwards (both 32 and 64 bit).

The .dlls and .infs show up in system32 and sysWOW64, the following registries are set correctly:

```
"HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\MediaResources\icm\VIDC.H264\Driver" set to "TestAMFVFW.DLL"
"HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Drivers32\VIDC.H264" set to "TestAMFVFW.DLL"
```

I've tried manually running both the .infs from their respective installed folders, nada.

The runtimes-registry doesn't exist, but since there's no more runtimes folder in the latest version, so I am assuming that registry entry is no longer needed?

I honestly don't know what else it could be, so if anyone could help, I'd appreciate it.

**Comment by jackun** [comment]
Ugh, this has been a very frustrating issue to debug. You can try [dependency walker](http://www.dependencywalker.com/) to see if it _does_ find all the (top-level) DLLs it links to.

> The runtimes-registry doesn't exist, but since there's no more runtimes folder in the latest version, so I am assuming that registry entry is no longer needed?

Indeed, just needs fairly new drivers.

Edit: Or just run config dialog manually: `rundll32 testamfvfw.dll,Configure`

**Comment by GunStingerNL** [comment]
According to Dependency Walker, I'm missing `D3DCOMPILER_47.DLL` which, according to Google is a DirectX SDK file which isn't included in Windows.

I also can't run the config dialog - I'm assuming I have to execute that command from the commandline and inside the system32/syswow64 folder, but both tell me `The specified module could not be found`

Edit: I downloaded the DX SDK for Win8.1, and nabbed the required DLL from there - now all the dependencies are found, but the install still doesn't work, nothing found in Sherlock.

Edit2: now the error to manually config says `testamfvfw.dll is not a valid Win32 application`

**Comment by jackun** [comment]
Hm should make it load d3dcompiler dynamically as it is used on win8+ only anyway.
See if [web setup](https://www.microsoft.com/en-us/Download/confirmation.aspx?id=35) installs it.

In edit2 you may have copied 64bit DLL to syswow64 or vice versa.

Edit: try the new binaries, should only load d3dcompiler on win8+ now.

**Comment by GunStingerNL** [comment]
Well, there's some progress: I got the manual config window to work now, but the codec still doesn't show up in Sherlock or DXTory...

**Comment by CharliBrwn** [comment]
I've got a solution and it works on my machine. Run the CMD as an admin, go manually to the directory where in install.bat is an run it. The codec should be installed after that :)

**Comment by GunStingerNL** [comment]
@CharliBrwn - I had already tried that, and that didn't work.

@jackun - I think I managed to fix the problem. Apparently there were still remnants of OpenEncodeVFW in my registry even after uninstalling it ages ago, clogging up a couple of `VIDC.H264` entries with the OpenEncode DLL. I manually removed all of them, then tried reinstalling the codec, and now it seems to be working as it should!
